File: e2e-travel-concierge/agent/chat/providers/openaiProvider.ts
import OpenAI from 'openai';
import { env } from '../../../config.js';
import type { LLMMessage, LLMProvider, LLMResponse } from './types.js';
import { chatComplete } from './openaiShared.js';

const MODEL = 'gpt-4o';

export class OpenAIProvider implements LLMProvider {
  private client = new OpenAI({ apiKey: env.llmApiKeyOrThrow() });

  complete(messages: LLMMessage[]): Promise<LLMResponse> {
    return chatComplete(this.client, MODEL, messages);
  }
}

File: examples/e2e-travel-concierge-py/agent/chat/providers/openai_provider.py
from __future__ import annotations

from typing import Any, List

from openai import OpenAI

from agent.chat.providers.openai_shared import chat_complete
from agent.chat.types import LLMMessage
from config import env

MODEL = "gpt-4o"


class OpenAIProvider:
    def __init__(self) -> None:
        self._client = OpenAI(api_key=env.llm_api_key())

    def complete(self, messages: List[LLMMessage]) -> Any:
        return chat_complete(self._client, MODEL, messages, "openai")
